### PR TITLE
Feat/connect core in popup

### PR DESCRIPTION
**Pull request by mroz22**
part of #9525 

builds on top of #9896 

this PR loads connect-core in popup. this will be useful for:
- support of env where iframe can't be loaded (service worker)
- support for webusb transport

### Commits:
- [de7789f](https://github.com/trezor/trezor-suite/pull/9914/commits/de7789fd996c5ee869a392fbf0cc7bc28ddea433) Ignore, this has its own PR in #9896 
- [4578f31](https://github.com/trezor/trezor-suite/pull/9914/commits/4578f315e1bac1db9db98e232c881eb0400fdf02) - new events for connect-webextension (#9525), they are not sent yet but popup already knows them
- [7e06e91](https://github.com/trezor/trezor-suite/pull/9914/commits/7e06e91e27ad5c2081096a8cdc228fd1410dcad0) - this is basically the same thing  `TRANSPORT.REQUEST_DEVICE` message does when communication goes through popup
- [3f002e2](https://github.com/trezor/trezor-suite/pull/9914/commits/3f002e2e30dccc009fb2a74ebd0ebee3017364cd) - this commit implements connect core in popup

### CODE REVIEW

File: packages/connect-popup/src/index.tsx [hunk 1]
```diff
@@ -186,8 +193,21 @@ const handleInitMessage = (event: MessageEvent<PopupEvent | IFrameLogRequest>) =
...
+            settings: {
+                ...getState().settings!,
+                origin: data.payload.id,
+            },
+        });
+        reactEventBus.dispatch({ type: 'state-update', payload: getState() });
+        return;
+    }
+
     if (data.type === IFRAME.LOG) {
```
Comment (karliatto) [inline packages/connect-popup/src/index.tsx:209]: This is for logs comming from service worker?

File: packages/connect-popup/src/view/selectDevice.ts [hunk 2]
```diff
@@ -32,16 +30,19 @@ const initWebUsbButton = (showLoader: boolean) => {
...
+            if (iframe) {
+                iframe.postMessage({
+                    event: UI_EVENT,
+                    type: TRANSPORT.REQUEST_DEVICE,
+                });
+                if (showLoader) {
+                    showView('loader');
+                }
+            } else if (core) {
+                core.enumerate();
```
Comment (karliatto) [inline packages/connect-popup/src/view/selectDevice.ts:45]: If there is no iframe nor core there is no error here as there was before?

File: packages/connect-popup/src/index.tsx [hunk 3]
```diff
@@ -13,10 +13,15 @@ import {
...
     MethodResponseMessage,
+    CORE_EVENT,
+    IFrameCallMessage,
     IFrameLogRequest,
     CoreMessage,
 } from '@trezor/connect/lib/exports';
+import { initLog } from '@trezor/connect/lib/utils/debug';
+import type { Core } from '@trezor/connect/lib/core';
 import { config } from '@trezor/connect/lib/data/config';
+import { parseConnectSettings } from '@trezor/connect-iframe/src/connectSettings';
```
Comment (karliatto) [inline packages/connect-popup/src/index.tsx:24]: Just a note or question, probably not related to this PR, but shouldn't this `parseConnectSettings` be in connect-common?

File: packages/connect-popup/src/index.tsx (hunk 1 above)
Comment (mroz22) [reply packages/connect-popup/src/index.tsx:209]: yes. I can't cross origin communicate from 3rd party service worker to shraredworekr on connect.trezor.io so it is done like this
maybe npm package should open log.html directly, inject script into it directly and communicate directly. this way we lose logs after popup is closed. but this is imho next step

File: packages/connect-popup/src/index.tsx (hunk 3 above)
Comment (mroz22) [reply packages/connect-popup/src/index.tsx:24]: imho no. there are about 3 modules `parseConnectSettings`. each one is located to a place which suggests level of how strict this validation should be. in this case, it should be the same strength like when using iframe. so I decided to keep it there for now, but who knows

File: packages/connect-popup/src/view/selectDevice.ts (hunk 2 above)
Comment (mroz22) [reply packages/connect-popup/src/view/selectDevice.ts:45]: yeah, maybe I put the error back

File: packages/connect-popup/webpack/prod.webpack.config.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import path from 'path';
 import webpack, { DefinePlugin } from 'webpack';
 import HtmlWebpackPlugin from 'html-webpack-plugin';
-import TerserPlugin from 'terser-webpack-plugin';
+// import TerserPlugin from 'terser-webpack-plugin';
```
Comment (mroz22) [inline packages/connect-popup/webpack/prod.webpack.config.ts]: this is temporary as well. I needed to check of code looks unminimized. removing it now

**Review by karliatto — APPROVED** [review]
Looks good to me.

**Pull request merged**